### ISSUE TITLE
Error in production mode

### ISSUE THREAD

**Issue by StefanKoenigMUC** [kmanaseryan/ngx-envconfig#4]
Hello,

while using the ngx-envconfig library, i can't use the production mode due to an error during application startup:

`production
main.91723b488af13390246c.js:1 TypeError: Cannot read property 'HOST_API' of undefined
    at t.getApi (main.91723b488af13390246c.js:1)
    at t.intercept (main.91723b488af13390246c.js:1)
    at t.handle (main.91723b488af13390246c.js:1)
    at t.intercept (main.91723b488af13390246c.js:1)
    at t.handle (main.91723b488af13390246c.js:1)
    at t.handle (main.91723b488af13390246c.js:1)
    at e.project (main.91723b488af13390246c.js:1)
    at e._tryNext (main.91723b488af13390246c.js:1)
    at e._next (main.91723b488af13390246c.js:1)
    at e.next (main.91723b488af13390246c.js:1)`

do you have an idea how i can fix this error?

thanks in advance,

**Comment by kmanaseryan** [comment]
@StefanKoenigMUC have you created `production.json` file under `assets/config` folder? 
You can find more [here in README](https://github.com/kmanaseryan/ngx-envconfig#setting-up-configuration-files) 

If you still have the issue please provide the file structure of `/environments` folder, how did you use the module in your app? and `.angular-cli.json` file so I will be able to have better understanding of your issue.

**Comment by StefanKoenigMUC** [comment]
files are available, it seems to me that this is some kind of timing problem during the startup. (As the config service is injected in other early used services, it seems that the config service is not always fully initialized). It's kind of hard to reproduce.

**Comment by kmanaseryan** [comment]
Have you included `ConfigModule` in imports array and called `.forRoot()` function?

```
// src/app/app.module.ts
import { NgModule } from '@angular/core';
import { ConfigModule, ConfigService } from './config/config.service';
 
@NgModule({
    imports: [
        ConfigModule.forRoot({state: 'development'})
        ...
    ],
    providers: [
        ...
        Your Providers
        ...
    ]
})
 
export class AppModule { }
```

**Comment by StefanKoenigMUC** [comment]
Yes, i've.  As in non production builds everything is working fine. Since one of the latest updates its working :-) Thanks for your help.

But now i'm getting another error: 

`ERROR in Error during template compile of 'AppModule'
  Function calls are not supported in decorators but 'ConfigModule' was called in 'ɵ0'
    'ɵ0' calls 'ConfigModule'.`

I've no clue what angular is going to tell me by that. 

ConfigModule import looks like that:

`    ConfigModule.forRoot({ state: environment.production ? 'production' : 'development' }),
`
(it's mostly your provided example)

**Comment by kmanaseryan** [comment]
This is fixed in the latest release v1.2.2 and published on NPM registry too. The issue was related to [this line](https://github.com/kmanaseryan/ngx-envconfig/commit/9ea5ada2aa7aefca9ff4b07c71ea171968f0e296#diff-aeb3606d829d44e8a67075386dbc6873L43) 

Thanks for noticing it!

**Comment by kmanaseryan** [comment]
@StefanKoenigMUC let me know how it works on your end. If something happen feel free to reopen this issue or open new ones.
